cpp/src/aztec/honk/composer/composer_helper/composer_helper.cpp

```diff
@@ -0,0 +1,357 @@
+        return circuit_proving_key;
+    }
+    // Compute q_l, q_r, q_o, etc polynomials
+    ComposerHelper::compute_proving_key_base(circuit_constructor, ComposerType::STANDARD);
+
+    // Compute sigma polynomials (we should update that late)
+    // TODO: Update this
+    // compute_sigma_permutations<3, false>(circuit_proving_key.get());
+
+    return circuit_proving_key;
+}
+
+/**
+ * Compute verification key consisting of selector precommitments.
+ *
+ * @return Pointer to created circuit verification key.
+ * */
+template <typename CircuitConstructor>
+std::shared_ptr<verification_key> ComposerHelper::compute_verification_key(CircuitConstructor& circuit_constructor)
+{
+    if (circuit_verification_key) {
+        return circuit_verification_key;
+    }
+    if (!circuit_proving_key) {
+        compute_proving_key();
+    }
+
+    circuit_verification_key = ComposerHelper::compute_verification_key_base(
+        circuit_constructor, circuit_proving_key, crs_factory_->get_verifier_crs());
+    circuit_verification_key->composer_type = circuit_proving_key->composer_type;
+
+    return circuit_verification_key;
+}
+
+/**
+ * Create verifier: compute verification key,
+ * initialize verifier with it and an initial manifest and initialize commitment_scheme.
+ *
+ * @return The verifier.
+ * */
+template <typename CircuitConstructor> Verifier ComposerHelper::create_verifier(CircuitConstructor& circuit_constructor)
+{
+    compute_verification_key(circuit_constructor);
+    // TODO figure out types, actuallt
+    // circuit_verification_key->composer_type = type;
+
+    // TODO: initialize verifier according to manifest and key
+    // Verifier output_state(circuit_verification_key, create_manifest(public_inputs.size()));
+    Verifier output_state;
+    // TODO: Do we need a commitment scheme defined here?
+    // std::unique_ptr<KateCommitmentScheme<standard_settings>> kate_commitment_scheme =
+    //    std::make_unique<KateCommitmentScheme<standard_settings>>();
+
+    // output_state.commitment_scheme = std::move(kate_commitment_scheme);
+
+    return output_state;
+}
+
+template <typename CircuitConstructor>
+UnrolledVerifier ComposerHelper::create_unrolled_verifier(CircuitConstructor& circuit_constructor)
+{
+    compute_verification_key(circuit_constructor);
+    // UnrolledVerifier output_state(circuit_verification_key,
+    //                             create_unrolled_manifest(circuit_constructor.public_inputs.size()));
+    UnrolledVerifier output_state;
+
+    // TODO: Deal with commitments
+    // std::unique_ptr<KateCommitmentScheme<unrolled_standard_settings>> kate_commitment_scheme =
+    //     std::make_unique<KateCommitmentScheme<unrolled_standard_settings>>();
+
+    // output_state.commitment_scheme = std::move(kate_commitment_scheme);
+
+    return output_state;
+}
+
+template <typename CircuitConstructor>
+UnrolledProver ComposerHelper::create_unrolled_prover(CircuitConstructor& circuit_constructor)
```

Comment:
I think this could be removed altogether if you want. I don't think there will be any notion of an 'unrolled' Honk Prover. Fine to leave it for now pending the unification of all of this stuff for Plonk/Honk.